pkg/config/validation/validation.go

```diff
@@ -2085,6 +2093,20 @@ func validateHTTPRoute(http *networking.HTTPRoute, delegate bool) (errs error) {
 	return
 }
 
+func validateStringMatchRegexp(sm *networking.StringMatch, where string) error {
+	re := sm.GetRegex()
+	if re == "" {
+		return nil
+	}
+
+	_, err := regexp.Compile(re)
+	if err == nil {
+		return nil
+	}
+
+	return fmt.Errorf("%s regexp invalid: %w", where, err)
```

Comment:
Nit: maybe reference a doc or something indicating we use re2 regex?

Reply:
I put a `hold` on to address this.  The error messages from `regexp.Compile()` are nice.  I ran this using `istioctl validate -f galley/pkg/config/analysis/analyzers/testdata/virtualservice_regexes.yaml` and the messages are:
```
	* VirtualService//bad-match: uri regexp invalid: error parsing regexp: missing closing ]: `[A-Z`
	* VirtualService//ecma-not-v2: uri regexp invalid: error parsing regexp: invalid or unsupported Perl syntax: `(?!`
```

Should it say something like 
```
VirtualService//ecma-not-v2: "uri": invalid or unsupported Perl syntax: `(?!`; Istio uses RE2 style regex-based match (https://github.com/google/re2/wiki/Syntax)
```
?